app/controllers/posts_controller.rb

```diff
@@ -37,4 +45,12 @@ def create
   def post_params
     params.require(:post).permit(:title, :text)
   end
+
+  def destroy
+    @post = Post.find(params[:id])
+    @author = @post.author
+    @author.decrement!(:post_counter)
+    @post.destroy!
+    redirect_to user_posts_path(id: @author.id), notice: 'The post has been successfully removed!'
+  end
```

Comment:
- [ ] Good job by adding the destroy action for the post and the button to execute it but when I try to delete a post I get an error. You should make sure when you delete a post everything should be destroyed even likes_counter, please kindly fix that 🙏🏾 

<img width="759" alt="Screenshot 2023-11-10 at 14 09 39" src="https://github.com/Marcraphael12/rails-blog-app/assets/74501523/f9530d5b-7fc9-453e-8f45-11c3953a5c8b">